core/store/models/common.go

```diff
@@ -425,20 +425,41 @@ func (c Cron) String() string {
 	return string(c)
 }
 
-// Duration is a time duration.
-type Duration time.Duration
+// Duration is a non-negative time duration.
+type Duration struct{ d time.Duration }
+
+func NewDuration(d time.Duration) (Duration, error) {
+	if d < time.Duration(0) {
+		return Duration{}, fmt.Errorf("cannot make negative time duration: %s", d)
+	}
+	return Duration{d: d}, nil
+}
+
+func MustNewDuration(d time.Duration) Duration {
+	rv, err := NewDuration(d)
+	if err != nil {
+		panic(err)
+	}
+	return rv
+}
 
 // Duration returns the value as the standard time.Duration value.
 func (d Duration) Duration() time.Duration {
-	return time.Duration(d)
+	return d.d
 }
 
+// Shorter returns true iff d is shorter than od.
```

Comment:
`iff` is that a typo?

Reply:
Short for "if and only if". I will expand it.